crates/lox-io/src/ndm/kvn/parser.rs

```diff
@@ -0,0 +1,1013 @@
+    }
+}
+
+impl<I> From<nom::Err<KvnDateTimeParserErr<I>>> for KvnDeserializerErr<I> {
+    fn from(value: nom::Err<KvnDateTimeParserErr<I>>) -> Self {
+        match value {
+            nom::Err::Error(KvnDateTimeParserErr::EmptyValue { input })
+            | nom::Err::Failure(KvnDateTimeParserErr::EmptyValue { input }) => {
+                KvnDeserializerErr::EmptyValue { input }
+            }
+            nom::Err::Error(KvnDateTimeParserErr::EmptyKeyword { input })
+            | nom::Err::Failure(KvnDateTimeParserErr::EmptyKeyword { input }) => {
+                KvnDeserializerErr::EmptyKeyword { input }
+            }
+            nom::Err::Error(KvnDateTimeParserErr::InvalidFormat { input })
+            | nom::Err::Failure(KvnDateTimeParserErr::InvalidFormat { input }) => {
+                KvnDeserializerErr::InvalidDateTimeFormat { input }
+            }
+            nom::Err::Error(KvnDateTimeParserErr::ParserError(i, k))
+            | nom::Err::Failure(KvnDateTimeParserErr::ParserError(i, k)) => {
+                KvnDeserializerErr::GeneralParserError(i, k)
+            }
+            // We don't use streaming deserialization
+            nom::Err::Incomplete(_) => unimplemented!(),
+        }
+    }
+}
+
+impl<I> From<nom::Err<KvnNumberParserErr<I>>> for KvnDeserializerErr<I> {
+    fn from(value: nom::Err<KvnNumberParserErr<I>>) -> Self {
+        match value {
+            nom::Err::Error(KvnNumberParserErr::EmptyValue { input })
+            | nom::Err::Failure(KvnNumberParserErr::EmptyValue { input }) => {
+                KvnDeserializerErr::EmptyValue { input }
+            }
+            nom::Err::Error(KvnNumberParserErr::EmptyKeyword { input })
+            | nom::Err::Failure(KvnNumberParserErr::EmptyKeyword { input }) => {
+                KvnDeserializerErr::EmptyKeyword { input }
+            }
+            nom::Err::Error(KvnNumberParserErr::InvalidFormat { input })
+            | nom::Err::Failure(KvnNumberParserErr::InvalidFormat { input }) => {
+                KvnDeserializerErr::InvalidDateTimeFormat { input }
+            }
+            nom::Err::Error(KvnNumberParserErr::ParserError(i, k))
+            | nom::Err::Failure(KvnNumberParserErr::ParserError(i, k)) => {
+                KvnDeserializerErr::GeneralParserError(i, k)
+            }
+            // We don't use streaming deserialization
+            nom::Err::Incomplete(_) => unimplemented!(),
+        }
+    }
+}
+
+impl<I> From<KvnKeyMatchErr<I>> for KvnDeserializerErr<I> {
+    fn from(value: KvnKeyMatchErr<I>) -> Self {
+        match value {
+            KvnKeyMatchErr::KeywordNotFound { expected } => {
+                KvnDeserializerErr::KeywordNotFound { expected }
+            }
+        }
+    }
+}
+
+impl<I> ParseError<I> for KvnStringParserErr<I> {
+    fn from_error_kind(input: I, kind: ErrorKind) -> Self {
+        KvnStringParserErr::ParserError(input, kind)
+    }
+
+    fn append(_: I, _: ErrorKind, other: Self) -> Self {
```

Comment:
Is this unused? I can't find a caller.

Reply:
It's part of the trait.

Reply:
Ah, I totally blanked that we were in a trait `impl` block. Never mind!